<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ReportsGeneratorSetupDialog</class>
 <widget class="QDialog" name="ReportsGeneratorSetupDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>692</width>
    <height>576</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0" colspan="4">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabParameters">
      <attribute name="title">
       <string>Параметры отчета</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QLineEdit" name="edtInputFileName"/>
       </item>
       <item row="0" column="1">
        <widget class="QToolButton" name="btnBrowseInputFile">
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QTableView" name="tblParameters">
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QComboBox" name="cmbTemplateFile"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabAdditionaOptions">
      <attribute name="title">
       <string>Дополнительные настройки</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="1" column="0">
        <widget class="QGroupBox" name="gbHeaderSection">
         <property name="title">
          <string>Верхняя (заголовочная) секция</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="4" column="1">
           <widget class="QSpinBox" name="spbHeaderColumnCount">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>8</number>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="spbHeaderRowCount">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>8</number>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="0" rowspan="6">
           <widget class="QTextEdit" name="tedtHeaderSection"/>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="lblHeaderRowCount">
            <property name="text">
             <string>строк:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="lblHeaderColumnCount">
            <property name="text">
             <string>стобцов:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QPushButton" name="btnAddToHeader">
            <property name="maximumSize">
             <size>
              <width>23</width>
              <height>23</height>
             </size>
            </property>
            <property name="text">
             <string>+</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QGroupBox" name="gbFooterSection">
         <property name="title">
          <string>Нижняя секция</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_9">
          <item row="0" column="0" rowspan="6">
           <widget class="QTextEdit" name="tedtFooterSection"/>
          </item>
          <item row="5" column="1">
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="lblFooterColumnCount">
            <property name="text">
             <string>стобцов:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="lblFooterRowCount">
            <property name="text">
             <string>строк:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="spbFooterColumnCount">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>8</number>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="spbFooterRowCount">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>8</number>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QPushButton" name="btnAddToFooter">
            <property name="maximumSize">
             <size>
              <width>23</width>
              <height>23</height>
             </size>
            </property>
            <property name="text">
             <string>+</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QGroupBox" name="gbMainTableSection">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>Основная таблица</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_10">
          <item row="1" column="0">
           <widget class="QTableWidget" name="mainTableColumns">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::DoubleClicked</set>
            </property>
            <property name="horizontalScrollMode">
             <enum>QAbstractItemView::ScrollPerPixel</enum>
            </property>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <row>
             <property name="text">
              <string>Новая строка</string>
             </property>
            </row>
            <column>
             <property name="text">
              <string>Новый столбец</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Новый столбец</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Новый столбец</string>
             </property>
            </column>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QCheckBox" name="chkTranspose">
            <property name="text">
             <string>Развернуть на 90 градусов против часовой стрелки</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="CTextAppearanceEditor" name="textAppearanceEditor" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabPreview">
      <attribute name="title">
       <string>Предпросмотр</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QTextEdit" name="tedtPreview"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Текст запроса</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="0">
        <widget class="QPlainTextEdit" name="tedtQueryStmt"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabInfo">
      <attribute name="title">
       <string>?</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QTextEdit" name="tedtInfo">
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:14px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt; font-weight:600;&quot;&gt;Общие положения&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Входной файл (далее &amp;quot;&lt;/span&gt;&lt;span style=&quot; font-size:8pt; font-weight:600;&quot;&gt;шаблон отчета&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;&amp;quot;) должен иметь расширение &amp;quot;sql&amp;quot;.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt; font-weight:600;&quot;&gt;Шаблон отчета&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt; может содержать: &lt;/span&gt;&lt;/p&gt;
&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:12px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;текст SQL-запроса для выборки данных из базы, &lt;/li&gt;
&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;комментарии в формате SQL, &lt;/li&gt;
&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;настраиваемые &lt;span style=&quot; font-weight:600;&quot;&gt;параметры отчета&lt;/span&gt;. &lt;/li&gt;&lt;/ul&gt;
&lt;p style=&quot; margin-top:14px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt; font-weight:600;&quot;&gt;Параметры отчета&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Параметр отчета представляет собой строку вида: {&lt;/span&gt;&lt;span style=&quot; font-size:8pt; font-style:italic;&quot;&gt;name&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;[#&lt;/span&gt;&lt;span style=&quot; font-size:8pt; font-style:italic;&quot;&gt;n&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;]@&lt;/span&gt;&lt;span style=&quot; font-size:8pt; font-style:italic;&quot;&gt;type&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;} &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;, где: &lt;/span&gt;&lt;/p&gt;
&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:12px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;name&lt;/span&gt; - имя параметра. Может содержать буквы и символы подчеркивания (последние будут заменятся на пробемы при отображении названия). &lt;/li&gt;
&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;n&lt;/span&gt; - сортировочный индекс. Указывает порядок вывода параметра в списке. Может быть опущен. &lt;/li&gt;
&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;type&lt;/span&gt; - тип параметра. Определяет используемый для редактирования элемент интерфейса.&lt;/li&gt;&lt;/ul&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:1; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;. &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;например, он может выглядеть так: {&lt;/span&gt;&lt;span style=&quot; font-size:8pt; font-style:italic;&quot;&gt;Дата_рождения_пациента&lt;/span&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;#&lt;/span&gt;&lt;span style=&quot; font-size:8pt; font-style:italic;&quot;&gt;2@date} &lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Поддерживаемые типы: &lt;/span&gt;&lt;/p&gt;
&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:12px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;date&lt;/span&gt; - для редактирования используется элемент с календарем. &lt;/li&gt;
&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;datetime&lt;/span&gt; - то же, что и date, но с возможностью указывать еще и время. &lt;/li&gt;
&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;str&lt;/span&gt; - для редактирования используется обычное поле ввода текста.&lt;/li&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:1; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;. &lt;/span&gt;&lt;/p&gt;
&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;int&lt;/span&gt; - для редактирования используется spinbox с пределами 0 - 999 и шагом 1. &lt;/li&gt;
&lt;li style=&quot; font-size:8pt;&quot; style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;float&lt;/span&gt; - для редактирования используется spinbox с пределами 0.0 - 99.9 и шагом 0.1. &lt;/li&gt;&lt;/ul&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QPushButton" name="btnSaveChanged">
     <property name="text">
      <string>Сохранить изменения</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>269</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="3">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QCheckBox" name="chkIsLandscapeOrientation">
     <property name="text">
      <string>Альбомная ориентация</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CTextAppearanceEditor</class>
   <extends>QWidget</extends>
   <header>Reports/ReportsGenerator/TextAppearanceEditor.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ReportsGeneratorSetupDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>682</x>
     <y>566</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ReportsGeneratorSetupDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>682</x>
     <y>566</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gbHeaderSection</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbHeaderRowCount</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>62</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>659</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gbHeaderSection</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbHeaderColumnCount</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>322</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>659</x>
     <y>162</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gbFooterSection</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbFooterRowCount</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>219</x>
     <y>524</y>
    </hint>
    <hint type="destinationlabel">
     <x>659</x>
     <y>434</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gbFooterSection</sender>
   <signal>toggled(bool)</signal>
   <receiver>spbFooterColumnCount</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>335</x>
     <y>524</y>
    </hint>
    <hint type="destinationlabel">
     <x>659</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gbHeaderSection</sender>
   <signal>toggled(bool)</signal>
   <receiver>btnAddToHeader</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>177</x>
     <y>61</y>
    </hint>
    <hint type="destinationlabel">
     <x>624</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gbFooterSection</sender>
   <signal>toggled(bool)</signal>
   <receiver>btnAddToFooter</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>275</x>
     <y>388</y>
    </hint>
    <hint type="destinationlabel">
     <x>629</x>
     <y>496</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
